Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -0,0 +1,583 @@
+        public IEnumerable<LookupViewModel> GetSeriesLookup(string type)
+        {
+            if (type == "status")
+            {
+                var res = _context.SeriesStatus
+                    .Select(s => new LookupViewModel
+                    {
+                        ID = s.ID,
+                        Description = s.Description
+                    })
+                    .AsEnumerable();
+                return res;
+            }
+            if (type == "series")
+            {
+                var res = _context.SeriesType
+                    .Select(s => new LookupViewModel
+                    {
+                        ID = s.ID,
+                        Description = s.Description
+                    })
+                    .AsEnumerable();
+                return res;
+            }
+            return null;
+        }
+        public IEnumerable<SeriesExtendedViewModel> GetSeries(bool active = false)
+        {
+            var series = _context.Series
+                            .Select(s => new SeriesExtendedViewModel
+                            {
+                                ID = s.ID,
+                                Name = s.Name,
+                                StartDate = s.StartDate,
+                                EndDate = s.EndDate,
+                                Type = _context.SeriesType
+                                            .FirstOrDefault(st => st.ID == s.TypeID)
+                                            .Description,
+                                Status = _context.SeriesStatus
+                                            .FirstOrDefault(ss => ss.ID == s.StatusID)
+                                            .Description,
+                                ActivityID = s.ActivityID,
+                                Match = _matchService.GetMatchBySeriesID(s.ID),
+                                TeamStanding = _context.SeriesTeam
+                                .Where(st => st.SeriesID == s.ID)
+                                .Select(st => new TeamRecordViewModel
+                                {
+                                    ID = st.ID,
+                                    Name = _context.Team
+                                            .FirstOrDefault(t => t.ID == st.TeamID)
+                                            .Name,
+                                    Win = st.Win,
+                                    Loss = st.Loss,
+                                    Tie = _context.SeriesTeam
+                                            .Where(total => total.TeamID == st.TeamID && total.SeriesID == s.ID)
+                                            .Count() - st.Win - st.Loss
+                                }).OrderByDescending(st => st.Win).AsEnumerable()
+                            });
+            if (active)
+            {
+                series = series.Where(s => s.StartDate < DateTime.Now
+                                        && s.EndDate > DateTime.Now);
```

Comment:
Theis is currently not inclusive of EndDate, did you mean it to be? I think the most common way to refer to date spans like this is with an inclusive start and end, not a mix of inclusive and exclusive.

Reply:
The main thing about the end dates are that they are actually specifically datetimes. Meaning that if something were to end on May 4th 00:00, it would last till May 3rd 23:59:59. It's more clear on the form where the actual post is coming from. But I'll leave this issue open for further discussion.

Reply:
Why don't you just compare the date values instead of datetime values? Then you could do `>=` and `<=`, handling the inclusivity of the end date pretty easily.

Reply:
I would add an issue for this. The logic can get confusing pretty fast if you're mixing inclusivity and exclusivity in date spans.

Reply:
Not going to lie, I don't think we even have a Series End condition that we use to compare to anything. As well as no strict reinforcement of the column. I think we might need to think of a bit of rework here.